### PR TITLE
Add yml for msal automation nightly pipeline in Engineering project

### PR DESCRIPTION
**Pull request by shahzaibj**
Move the MSAL Automation Nightly Pipeline from IDDP to Engineering project and also convert to YML.

This pipeline effectively replaces the following pipeline in IDDP: https://dev.azure.com/IdentityDivision/IDDP/_build?definitionId=1237

### CODE REVIEW

**Review by p3dr0rv — APPROVED** [review]

File: azure-pipelines/ui-automation/msal-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# run broker UI automation testcases
```
Comment (somalaya) [inline azure-pipelines/ui-automation/msal-nightly.yml]: nit : should this be run msal automation testcases?

Comment (shahzaibj) [reply azure-pipelines/ui-automation/msal-nightly.yml]: yes

**Review by somalaya — APPROVED** [review]

File: azure-pipelines/ui-automation/msal-nightly.yml (hunk 1 above)
Comment (shahzaibj) [inline azure-pipelines/ui-automation/msal-nightly.yml]:
```suggestion
# run MSAL UI automation testcases
```

Comment (shahzaibj) [reply azure-pipelines/ui-automation/msal-nightly.yml]: Addressed here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1617/commits/35c3e05a87b8a8e00af8b61d6bbfc45656d0f2a2

**Pull request merged**